pkg/reconciler/revision/reconcile_resources.go

```diff
@@ -63,6 +63,17 @@ func (c *Reconciler) reconcileDeployment(ctx context.Context, rev *v1alpha1.Revi
 			logger.Errorf("Error updating deployment %q: %v", deploymentName, err)
 			return err
 		}
+
+		// Now that we have a Deployment, determine whether there is any relevant
+		// status to surface in the Revision.
+		//
+		// TODO(jonjohnsonjr): Should we check Generation != ObservedGeneration?
+		// The autoscaler mutates the deployment pretty often, which would cause us
+		// to flip back and forth between Ready and Unknown every time we scale up
+		// or down.
+		if !rev.Status.IsActivationRequired() {
```

Comment:
I kind of wonder if this should be folded into the propagation 🤔 

Why did this move?

Reply:
> I kind of wonder if this should be folded into the propagation

Possibly... I'm honestly not sure why we have this at all -- we're basically masking the underlying deployment status instead of exposing it. If we just exposed it, it seems like we'd have equivalent semantics (scaled to zero deployment is ready anyway, and we'd surface fatal errors faster if we didn't have to wait for activation to timeout...).

> Why did this move?

There might be a better way to do this. We're currently calling `MarkDeploying` before creating the deployment. This was overwriting that, so I moved PropagateDeploymentStatus into this else clause so that we don't call it when we first create the deployment.

I'm not sure if we care to keep the "Deploying" thing now that we have this, but I was trying to minimize the test impact.